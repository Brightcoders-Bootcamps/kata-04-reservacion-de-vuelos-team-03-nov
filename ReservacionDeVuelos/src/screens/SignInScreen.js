import React, {useContext, useState} from 'react';
import {View, Text, Image, StyleSheet, TouchableOpacity} from 'react-native';
import colors from '../utils/Colors';
import contentText from '../utils/Constants';
import FormButton from '../components/FormButton';
import FormInput from '../components/FormInput';
import {AuthContext} from '../navigation/AuthProvider';
import GoogleButton from '../components/GoogleButton';
import Checkbox from '../components/Checkbox';
//import eye from '../img/ojo.png';

const SignInScreen = ({navigation}) => {
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPasword] = useState('');
  const {register, googleLogIn} = useContext(AuthContext);

  return (
    <View style={styles.container}>
      <View style={styles.ContainerInput}>
        <Text style={styles.title}> {contentText.signUpTitle} </Text>
        <Text style={styles.formInputtext}> {contentText.firstName} </Text>

        <FormInput
          labelValue={name}
          onChangeText={(userName) => setName(userName)}
          secureTextEntry={false}
        />
        <Text style={styles.formInputtext}>{contentText.email} </Text>

        <FormInput
          labelValue={email}
          onChangeText={(userEmail) => setEmail(userEmail)}
          secureTextEntry={false}
        />

        <Text style={styles.formInputtext}> {contentText.password} </Text>

        <FormInput
          labelValue={password}
          onChangeText={(userPassword) => setPasword(userPassword)}
          secureTextEntry={true}
          source={true}
        />
      </View>

      <Text style={styles.restricted_password}>
        {contentText.restricted_password}
      </Text>

      <Checkbox />

      <FormButton
        buttonTitle={contentText.signUpTitle}
        onPress={() => register(email, password)}
        backgroundColor={colors.gray}
        color={colors.white}
      />

      <Text style={styles.textOr}>or</Text>

      <GoogleButton
        buttonTitle={contentText.signUpGoogle}
        onPress={() => googleLogIn()}
        backgroundColor={colors.gray}
        color={colors.white}
      />

      <View style={styles.login}>
        <Text style={styles.textLogIn}>{contentText.haveAccount}</Text>
        <TouchableOpacity onPress={() => navigation.navigate('LogIn')}>
          <Text style={styles.goToSignIn}> {contentText.logInTitle} </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.white,
  },
  title: {
    fontSize: 20,
    color: colors.blue,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  ContainerInput: {
    marginTop: 20,
    marginBottom: 10,
  },
  formInputtext: {
    color: colors.black,
    fontSize: 13,
    padding: 3,
  },
  goToSignIn: {
    color: colors.blue,
    fontSize: 16,
    textDecorationLine: 'underline',
  },
  login: {
    flexDirection: 'row',
    marginTop: 20,
    color: colors.black,
  },
  textLogIn: {
    fontSize: 16,
    color: colors.darkGray,
  },
  goToLogIn: {
    color: colors.blue,
  },
  restricted_password: {
    fontSize: 11,
    marginLeft: 16,
  },
  textOr: {
    color: colors.darkGray,
    marginTop: 5,
    marginBottom: 5,
  },
});

export default SignInScreen;
