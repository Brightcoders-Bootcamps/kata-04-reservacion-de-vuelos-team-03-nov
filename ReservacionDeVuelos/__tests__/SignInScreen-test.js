import React from 'react';
import renderer from 'react-test-renderer';
import SignInScreen from '../src/screens/SignInScreen';

test('renders correctly', () => {
  const tree = renderer.create(<SignInScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
