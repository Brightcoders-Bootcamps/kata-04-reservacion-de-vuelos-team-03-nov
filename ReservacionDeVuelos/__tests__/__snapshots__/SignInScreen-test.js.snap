// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#046BF1",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 10,
          "marginLeft": -5,
        }
      }
    >
       
      Sign Up
       
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 13,
        }
      }
    >
       
      First Name
       
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderRadius": 3,
          "borderWidth": 1,
          "height": 88.93333333333334,
          "marginBottom": 10,
          "marginTop": 5,
          "width": 332,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        rejectResponderTermination={true}
        secureTextEntry={false}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#000",
            "color": "#333",
            "flex": 1,
            "fontSize": 16,
            "justifyContent": "flex-start",
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 13,
        }
      }
    >
      Email *
       
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderRadius": 3,
          "borderWidth": 1,
          "height": 88.93333333333334,
          "marginBottom": 10,
          "marginTop": 5,
          "width": 332,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        rejectResponderTermination={true}
        secureTextEntry={false}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#000",
            "color": "#333",
            "flex": 1,
            "fontSize": 16,
            "justifyContent": "flex-start",
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 13,
        }
      }
    >
       
      Password *
       
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderRadius": 3,
          "borderWidth": 1,
          "height": 88.93333333333334,
          "marginBottom": 10,
          "marginTop": 5,
          "width": 332,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#000",
            "color": "#333",
            "flex": 1,
            "fontSize": 16,
            "justifyContent": "flex-start",
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
          "width": "20%",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 30,
            "marginLeft": 280,
            "marginTop": -60,
            "width": 30,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 11,
        "marginLeft": 16,
        "marginTop": -10,
      }
    }
  >
    Use 8 or more characteres  with a mix of letters, numbers, and symbols.
  </Text>
  <View
    style={
      Object {
        "marginHorizontal": 2,
        "marginLeft": -18,
        "marginTop": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 5,
          "width": "90%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
          tintColors={
            Object {
              "false": "#000",
              "true": "#046BF1",
            }
          }
          value={false}
        />
      </View>
      <Text
        style={
          Object {
            "color": "#757575",
            "fontSize": 13,
            "margin": 8,
          }
        }
      >
        I agree to the
        <Text
          style={
            Object {
              "textDecorationLine": "underline",
            }
          }
        >
           Terms
        </Text>
         and
        <Text
          style={
            Object {
              "textDecorationLine": "underline",
            }
          }
        >
          Privacy Policy.
        </Text>
        *
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 5,
          "width": "90%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
          tintColors={
            Object {
              "false": "#000",
              "true": "#046BF1",
            }
          }
          value={false}
        />
      </View>
      <Text
        style={
          Object {
            "color": "#757575",
            "fontSize": 13,
            "margin": 8,
          }
        }
      >
         
        Subscribe for select product updates.
         
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#C2C4C0",
        "borderRadius": 10,
        "height": 88.93333333333334,
        "justifyContent": "center",
        "marginTop": 10,
        "opacity": 1,
        "padding": 10,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Sign Up
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#757575",
        "marginBottom": 5,
        "marginTop": 5,
      }
    }
  >
    or
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#C2C4C0",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 88.93333333333334,
        "marginTop": 10,
        "opacity": 1,
        "padding": 10,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "20%",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Sign Up with Google
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "color": "#000",
        "flexDirection": "row",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#757575",
          "fontSize": 16,
        }
      }
    >
      Already have an account?
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#046BF1",
            "fontSize": 16,
            "textDecorationLine": "underline",
          }
        }
      >
         
        Log In
         
      </Text>
    </View>
  </View>
</View>
`;
